mne/epochs.py

```diff
@@ -682,272 +799,47 @@ def plot_psd_topomap(self, bands=None, vmin=None, vmax=None, proj=False,
-                if on_missing == 'error':
-                    raise ValueError(msg)
-                elif on_missing == 'warning':
-                    logger.warning(msg)
-                    warnings.warn(msg)
-                else:  # on_missing == 'ignore':
-                    pass
-
-        # Select the desired events
-        values = list(self.event_id.values())
-        selected = in1d(events[:, 2], values)
-        self.events = events[selected]
-
-        n_events = len(self.events)
-        if n_events > 1:
-            if np.diff(self.events.astype(np.int64)[:, 0]).min() <= 0:
-                warnings.warn('The events passed to the Epochs constructor '
-                              'are not chronologically ordered.',
-                              RuntimeWarning)
-
-        if n_events > 0:
-            logger.info('%d matching events found' % n_events)
-        else:
-            raise ValueError('No desired events found.')
-
-        self.selection = np.where(selected)[0]
-        self.drop_log = []
-        for k in range(len(events)):
-            if events[k, 2] in values:
-                self.drop_log.append([])
-            else:
-                self.drop_log.append(['IGNORED'])
-
-        self.preload = preload
-        if self.preload:
-            self._data = self._get_data_from_disk()
-            self.raw = None
-        else:
-            self._data = None
-
-    def drop_bad_epochs(self):
+    def drop_bad_epochs(self, reject='original', flat='original'):
         """Drop bad epochs without retaining the epochs data.
 
         Should be used before slicing operations.
 
         .. Warning:: Operation is slow since all epochs have to be read from
-            disk. To avoid reading epochs form disk multiple times, initialize
+            disk. To avoid reading epochs from disk multiple times, initialize
             Epochs object with preload=True.
 
+        Parameters
+        ----------
+        reject : dict | str | None
+            Rejection parameters based on peak-to-peak amplitude.
+            Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg'.
+            If reject is None then no rejection is done. If 'original',
+            then the rejection parameters set at instantiation are used.
+        flat : dict | str | None
+            Rejection parameters based on flatness of signal.
+            Valid keys are 'grad' | 'mag' | 'eeg' | 'eog' | 'ecg', and values
+            are floats that set the minimum acceptable peak-to-peak amplitude.
+            If flat is None then no rejection is done. If 'original',
+            then the flat parameters set at instantiation are used.
+
+        Notes
+        -----
+        Dropping bad epochs can be done multiple times with different
+        ``reject`` and ``flat`` parameters. However, once an epoch is
+        dropped, it is dropped forever, so if more lenient thresholds may
+        subsequently be applied, `epochs.copy` should be used.
```

Comment:
no no, not rechecking the thresholds. Short-circuiting is the way to go. But the user may want to use it a couple of times in their script just to make sure that they have dropped the epochs (in case they are not sure it was done in some other part of the script). But you want to keep the same thresholds, not change it. I think we agree?